### PR TITLE
3545 - 2

### PR DESCRIPTION
**Pull request by NikitaMasalkov**
## Номер
3545

## Номер задания
2

## Ссылка на видео с демо
https://recordit.co/USjjwe1Pe4

### CODE REVIEW

File: 3545/2/commands/checkin.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  def upload_geo(*)
+    File.open(checkin_path(chat['id']) + '/geo.txt', 'wb') do |file|
+      p payload
+      file << payload['location'].to_s
+    end
+    respond_with :message, text: 'Отлично, порви сегодня всех. За себя и за Сашку.'
+  end
+
+  def create_checkin_dir
+    FileUtils.mkdir_p(checkin_path)
```
Comment (anatoliliotych) [inline 3545/2/commands/checkin.rb]: are you creating each time? maybe it's better to check if it exists?

File: 3545/2/commands/checkin.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  end
+
+  def checkin_path
+    "public/#{chat['id']}/checkins/#{Time.now.strftime('%a, %d %b %Y %H:%M')}"
+  end
+
+  def image_path
+    JSON.parse(URI.open(API_URL + payload['photo'].last['file_id'])
```
Comment (anatoliliotych) [inline 3545/2/commands/checkin.rb]: this is unreadable

File: 3545/2/webhooks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+require 'telegram/bot'
+require 'logger'
+require 'active_support/all'
+require 'redis-rails'
+require 'csv'
+require './commands/start'
+require './commands/checkin'
+require './commands/checkout'
+
+TOKEN = '871720433:AAE0re0dH98iwJ98bBgL-14YaB8ZjvVdk9s'
```
Comment (anatoliliotych) [inline 3545/2/webhooks_controller.rb]: not safe!

File: 3545/2/commands/start.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    save_context :authenticate
+    message = session[:id] ? 'Вы уже зарегестрировались' : 'Введите номер по лагерю'
+
+    respond_with :message, text: message
+  end
+
+  def authenticate(response)
+    session[:response] = response
+
+    if CSV.read('./data.csv').flatten.include?(response) && session[:id] != response
```
Comment (anatoliliotych) [inline 3545/2/commands/start.rb]: unreadable

File: 3545/2/commands/checkout.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module CheckOutCommand
+  include Telegram::Bot::UpdatesController::MessageContext
+
+  def checkout!
+    session[:is_checkedin?] = false
+    FileUtils.mkdir_p(checkout_path)
```
Comment (anatoliliotych) [inline 3545/2/commands/checkout.rb]: the same question

File: 3545/2/commands/checkin.rb [hunk 6]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module CheckInCommand
+  TOKEN = '871720433:AAE0re0dH98iwJ98bBgL-14YaB8ZjvVdk9s'
```
Comment (anatoliliotych) [inline 3545/2/commands/checkin.rb]: not safe

File: 3545/2/commands/checkin.rb (hunk 1 above)
Comment (NikitaMasalkov) [reply 3545/2/commands/checkin.rb]: Sure, good point

File: 3545/2/commands/checkin.rb (hunk 2 above)
Comment (NikitaMasalkov) [reply 3545/2/commands/checkin.rb]: Divided in two methods

File: 3545/2/webhooks_controller.rb (hunk 3 above)
Comment (NikitaMasalkov) [reply 3545/2/webhooks_controller.rb]: Moved to env variable

File: 3545/2/commands/checkin.rb (hunk 6 above)
Comment (NikitaMasalkov) [reply 3545/2/commands/checkin.rb]: Moved to env variable

File: 3545/2/commands/checkin.rb (hunk 1 above)
Comment (anatoliliotych) [reply 3545/2/commands/checkin.rb]: why it's not fixed here?

**Pull request merged**